src/Bicep.LangServer/Handlers/BicepDidChangeWatchedFilesHandler.cs

```diff
@@ -1,30 +1,46 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Bicep.Core.FileSystem;
+using Bicep.Core;
+using Bicep.Core.Workspaces;
 using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Configuration;
 using MediatR;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using FileSystemWatcher = OmniSharp.Extensions.LanguageServer.Protocol.Models.FileSystemWatcher;
 
 namespace Bicep.LanguageServer.Handlers
 {
     public sealed class BicepDidChangeWatchedFilesHandler : DidChangeWatchedFilesHandlerBase
     {
+        private readonly IWorkspace workspace;
         private readonly ICompilationManager compilationManager;
-        private readonly IFileResolver fileResolver;
 
-        public BicepDidChangeWatchedFilesHandler(ICompilationManager compilationManager, IFileResolver fileResolver)
+        public BicepDidChangeWatchedFilesHandler(ICompilationManager compilationManager, IWorkspace workspace)
         {
             this.compilationManager = compilationManager;
-            this.fileResolver = fileResolver;
+            this.workspace = workspace;
         }
 
         public override Task<Unit> Handle(DidChangeWatchedFilesParams request, CancellationToken cancellationToken)
         {
-            compilationManager.HandleFileChanges(request.Changes);
+            Container<FileEvent> fileEvents = request.Changes;
+            IEnumerable<FileEvent> bicepConfigFileChangeEvents = fileEvents.Where(x => x.Uri.Path.EndsWith(LanguageConstants.BicepConfigSettingsFileName));
```

Comment:
>EndsWith [](http://example.com/codeflow?start=98&length=8)

`EndsWith()` by default will perform a case-sensitive comparison. On Windows, the file system is case-insensitive. Will there be a mismatch between this check and the logic that searches for `bicepconfig.json`?

I think it's ok for us to be case-sensitive even on a case-insensitive file system - I just want to make sure the comparisons are consistent with each other.

Reply:
Using case insensitive string comparison as part of this commit:
https://github.com/Azure/bicep/commit/1ee10711ad7bdd7412967c1ce6ae6f6a1923fa5b

Let me know if you have any concerns